ui/packages/catalog-ui-search/src/main/webapp/component/query-editor/query-editor.view.js

```diff
@@ -162,8 +160,7 @@ module.exports = Marionette.LayoutView.extend({
     const queryContentView = this.queryView
       ? this.queryView
       : this.queryContent.currentView
-    if (!queryContentView.isValid()) {
-      announcement.announce(InvalidSearchFormMessage)
+    if (!validate(queryContentView.validate())) {
```

Comment:
In terms of return value all your validate util does is check if the result of `queryContentView.validate()` is an empty array. I also don't think it reads well having two `validate`s in the same call. For these two reasons, I would recommend removing the return value, and renaming the `validate` within `utils/validation` to something like `announceErrors` so this line would look more like:
```
const errors = queryContentView.validate()
if (errors.length !== 0) {
    announceErrors(errors)    
    return
}
```

Reply:
The `validate` function inside of `utils/validation` also returns a boolean, `true` if the validation was successful, and `false` if it was unsuccessful. i think what you're suggesting is to move that check outside the validate function, which would mean refactoring to make this check each time the validate method is called. It might be a little clearer that way because then the `validate` (`announceErrors`) function would only be doing one thing rather than two, but this would also require adding some additional logic, for example, in `query-settings.view.js`, rather than doing this:

```
if (!validate(this.validate())) {
        return
      }
```

I would have to do something like this:

```
const errors = this.validate()   // this looks odd to me, and would probably result in having to rename this validate function as well
if(errors.length !== 0) {
    announceErrors(errors)
    return
}
```